consumer.go

```diff
@@ -108,15 +119,17 @@ func (c *segmentConsumer) Ack() error {
 	return nil
 }
 
-func (c *segmentConsumer) Close() {
+func (c *segmentConsumer) Close() error {
 	if c.reader == nil {
-		return
+		return nil
 	}
 	// this will also make the loops in the reader goroutines stop
 	err := c.reader.Close()
 	if err != nil {
-		sdk.Logger(context.Background()).Err(err).Msg("couldn't close reader")
+		return fmt.Errorf("couldn't close reader: %w", err)
```

Comment:
Not going to block the PR, but we should do the same in the `segmentProducer` when we get a chance.

Reply:
I'll do that here, so I don't forget about it. I was wondering why doesn't the linter warn about the error from Close() being ignored, but it looks like by default that warning is not shown for stuff from io.*.

Reply:
I was thinking the same thing, thanks for shedding some light onto this 👍